python/spec/dependabot/python/update_checker/index_finder_spec.rb

```diff
@@ -309,5 +309,46 @@
         end
       end
     end
+
+    describe "#environment_variables" do
+      subject do
+        finder.clean_check_and_remove_environment_variables(*arguments)
+      end
+
+      context "environment variable interpolation" do
+        let(:credentials) do
+          [
+            {
+              "type" => "python_index",
+              "index-url" => "https://${creds}@company.com/simple",
+              "token" => "user:password",
+              "replaces-base" => false
+            },
+            {
+              "type" => "python_index",
+              "index-url" => "https://${creds}@not.company.com/simple",
+              "token" => "user:password",
+              "replaces-base" => false
+            }
+          ]
+        end
+
+        it "selects the correct URL" do
+          let(:arguments) { "https://${creds}@company.com/simple" }
+
+          it do
+            is_expected.to eq "https://user:password@company.com/simple"
+          end
+        end
+
+        it "interpolates two variables" do
+          let(:arguments) { "https://${creds}@${domain}.company.com/simple" }
+
+          it do
+            is_expected.to eq "https://user:password@not.company.com/simple"
+          end
+        end
```

Comment:
Thanks for adding these tests 🙌 Could you also add a couple that shouldn't match?

For example: `https://${creds}@other.com/simple` and `https://${creds}@other.company.com/simple`

Reply:
Sure thing!